File: app/eclipse/futurehome/src/no/futurehome/futurehome_app/widget/models/ModeKt.java
package no.futurehome.futurehome_app.widget.models;

import kotlin.Metadata;
import kotlin.NoWhenBranchMatchedException;
import kotlin.jvm.internal.Intrinsics;

@Metadata(d1 = {"\000\016\n\000\n\002\020\016\n\000\n\002\030\002\n\000\032\016\020\000\032\0020\0012\006\020\002\032\0020\003¨\006\004"}, d2 = {"modeToString", "", "mode", "Lno/futurehome/futurehome_app/widget/models/Mode;", "app_release"}, k = 2, mv = {2, 1, 0}, xi = 48)
public final class ModeKt {
  public static final String modeToString(Mode paramMode) {
    String str;
    Intrinsics.checkNotNullParameter(paramMode, "mode");
    int i = WhenMappings.$EnumSwitchMapping$0[paramMode.ordinal()];
    if (i != 1) {
      if (i != 2) {
        if (i != 3) {
          if (i == 4) {
            str = "vacation";
          } else {
            throw new NoWhenBranchMatchedException();
          } 
        } else {
          str = "sleep";
        } 
      } else {
        str = "away";
      } 
    } else {
      str = "home";
    } 
    return str;
  }
}

File: app/futurehome-kotlin/no/futurehome/futurehome_app/widget/models/ModeKt.kt
package no.futurehome.futurehome_app.widget.models

public fun modeToString(mode: Mode): String {
   val var1: Int = ModeKt.WhenMappings.$EnumSwitchMapping$0[var0.ordinal()];
   val var2: java.lang.String;
   if (var1 != 1) {
      if (var1 != 2) {
         if (var1 != 3) {
            if (var1 != 4) {
               throw new NoWhenBranchMatchedException();
            }

            var2 = "vacation";
         } else {
            var2 = "sleep";
         }
      } else {
         var2 = "away";
      }
   } else {
      var2 = "home";
   }

   return var2;
}
// $VF: Class flags could not be determined
@JvmSynthetic
internal class WhenMappings {
   @JvmStatic
   public int[] $EnumSwitchMapping$0;

   @JvmStatic
   fun {
      val var0: IntArray = new int[Mode.values().length];

      try {
         var0[Mode.HOME.ordinal()] = 1;
      } catch (var5: NoSuchFieldError) {
      }

      try {
         var0[Mode.AWAY.ordinal()] = 2;
      } catch (var4: NoSuchFieldError) {
      }

      try {
         var0[Mode.SLEEP.ordinal()] = 3;
      } catch (var3: NoSuchFieldError) {
      }

      try {
         var0[Mode.VACATION.ordinal()] = 4;
      } catch (var2: NoSuchFieldError) {
      }

      $EnumSwitchMapping$0 = var0;
   }
}
